File: src/vision_node/vision_node/vision_monitor_node.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import Int32, Float64, String

class VisionMonitor(Node):

    def __init__(self):
        super().__init__('vision_monitor')

        #initialize self.latest_fps, self.latest_face_count,elf.latest_alert with sensible defaults
        self.latest_fps = 0.0
        self.latest_face_count = 0
        self.latest_alert = 'No alerts'


        self.subscriber_1 = self.create_subscription(Int32, '/vision/face_count', self.face_count_callback, 10)
        self.subscriber_2 = self.create_subscription(Float64, '/vision/fps', self.fps_callback, 10)
        self.subscriber_3 =self.create_subscription(String, '/vision/alerts', self.alert_callback, 10)

        self.timer = self.create_timer(1.0, self.print_status)

    def fps_callback(self, msg):
        self.latest_fps = msg.data

    def face_count_callback(self, msg):
        self.latest_face_count = msg.data

    def alert_callback(self, msg):
        self.latest_alert = msg.data

    def print_status(self):
        self.get_logger().info(f' [VISION STATUS] FPS: {self.latest_fps:.2f} | Faces: {self.latest_face_count} | Status: {self.latest_alert}')

def main(args=None):
    rclpy.init(args=args)
    node = VisionMonitor()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/vision_node/src/vision_monitor_node.cpp
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float64.hpp"
#include "std_msgs/msg/int32.hpp"
#include "std_msgs/msg/string.hpp"

class VisionMonitor : public rclcpp::Node
{
public:
  VisionMonitor()
  : Node("vision_monitor"),
    latest_fps_(0.0),
    latest_face_count_(0),
    latest_alert_("No alerts")
  {
    face_count_subscriber_ = this->create_subscription<std_msgs::msg::Int32>(
      "/vision/face_count", 10,
      std::bind(&VisionMonitor::face_count_callback, this, std::placeholders::_1));

    fps_subscriber_ = this->create_subscription<std_msgs::msg::Float64>(
      "/vision/fps", 10,
      std::bind(&VisionMonitor::fps_callback, this, std::placeholders::_1));

    alert_subscriber_ = this->create_subscription<std_msgs::msg::String>(
      "/vision/alerts", 10,
      std::bind(&VisionMonitor::alert_callback, this, std::placeholders::_1));

    timer_ = this->create_wall_timer(
      std::chrono::seconds(1),
      std::bind(&VisionMonitor::print_status, this));
  }

private:
  void fps_callback(const std_msgs::msg::Float64::SharedPtr msg)
  {
    latest_fps_ = msg->data;
  }

  void face_count_callback(const std_msgs::msg::Int32::SharedPtr msg)
  {
    latest_face_count_ = msg->data;
  }

  void alert_callback(const std_msgs::msg::String::SharedPtr msg)
  {
    latest_alert_ = msg->data;
  }

  void print_status()
  {
    RCLCPP_INFO(
      this->get_logger(),
      " [VISION STATUS] FPS: %.2f | Faces: %d | Status: %s",
      latest_fps_,
      latest_face_count_,
      latest_alert_.c_str());
  }

  rclcpp::Subscription<std_msgs::msg::Int32>::SharedPtr face_count_subscriber_;
  rclcpp::Subscription<std_msgs::msg::Float64>::SharedPtr fps_subscriber_;
  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr alert_subscriber_;
  rclcpp::TimerBase::SharedPtr timer_;

  double latest_fps_;
  int latest_face_count_;
  std::string latest_alert_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<VisionMonitor>());
  rclcpp::shutdown();
  return 0;
}
